### PR TITLE
Changelog dialog with crowdfunding announcement

### PR DESCRIPTION
**Pull request by signedav**
Looks like this

image

Opinions @m-kuhn @mbernasocchi ?

To do:
- [ ] Link and symbol in About
- [ ] Expire date

### CODE REVIEW

**Comment by 3nids** [comment]
Nice !
No remind me later?

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-5aa890b238607e2a85add3f752f05733a1464cdc-armv7-X.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-5aa890b238607e2a85add3f752f05733a1464cdc-x86-X.apk) (Android 5: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-5aa890b238607e2a85add3f752f05733a1464cdc-armv7-5.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-5aa890b238607e2a85add3f752f05733a1464cdc-x86-5.apk))

**Comment by mbernasocchi** [comment]
mmmm dos not look nice on my device.

Screenshot_20190924-181845_QField Beta 624
Screenshot_20190924-181822_QField Beta 624

even in the picture here I think there are some needed changes:
- [ ] More padding around the rectangle
- [ ] Rounded corners
- [ ] do not use black as the background. it is super hard
- [ ] do not use white as a border and background. it is super hard

PS: aren't this kind of things better developed in java with the premade android tools?

**Comment by signedav** [comment]
Ok. I'll fix the layout disaster tomorrow morning.

Regarding the popup it's similar to the other popups used in QField and I decided to make it in QML because its not Android related like Rating etc... (I talked with Mario about that).

What is for you the advantage of using java?

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-4106bf578d0cee57d6c618fc7f034e534378799d-armv7-X.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-4106bf578d0cee57d6c618fc7f034e534378799d-x86-X.apk) (Android 5: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-4106bf578d0cee57d6c618fc7f034e534378799d-armv7-5.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-4106bf578d0cee57d6c618fc7f034e534378799d-x86-5.apk))

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-9ab739bd0abccd7cc20553134753bbdaf451ec8a-armv7-X.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-9ab739bd0abccd7cc20553134753bbdaf451ec8a-x86-X.apk) (Android 5: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-9ab739bd0abccd7cc20553134753bbdaf451ec8a-armv7-5.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-9ab739bd0abccd7cc20553134753bbdaf451ec8a-x86-5.apk))

**Comment by mbernasocchi** [comment]
better in portrait mode. still some issues.
Screenshot_20190925-115123_QField Beta 624
the lower part looks good, the upper full width white  background should also be narrower like the lower part. also if possible use rounded edges.

in landscape mode it does not work properly. you can't scroll down. 
Screenshot_20190925-115615_QField Beta 624

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-1ed0679a1bcfd893ce85fb2ece8faa552716d865-armv7-X.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-1ed0679a1bcfd893ce85fb2ece8faa552716d865-x86-X.apk) (Android 5: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-1ed0679a1bcfd893ce85fb2ece8faa552716d865-armv7-5.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-1ed0679a1bcfd893ce85fb2ece8faa552716d865-x86-5.apk))

**Comment by nirvn** [comment]
Just compiled this branch. Here's how it looks on my computer screen:
Screenshot from 2019-09-26 15-38-45

Comments
- The word wrapping doesn't seem to work for list items, the text is being chopped off (see the first line on the relation editor widget).
- Generally speaking, a good layout tries to limit its font variant (family type, size, etc.), and in this case I'd argue there are two many varying sizes; I'd have two font size: one for the headline, and one for the rest (paragraphs, changelog widget, and buttons), and use a bold variant to highlight "support our crowdfunding..."
- The layout would benefit of some padding at the bottom of the headline, which would also balance the distribution of the elements below said headline.
- Headline in QField green, with a light drop shadow could look quite nice.
- I'd also add a top and bottom margin to the changelog list widget.

**Comment by signedav** [comment]
Alright @nirvn Thanks.
I try to follow your advises and push an update.

File: src/qml/Changelog.qml [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+            type: "New Feature"
+            description: "Value relation widget with multiple selection support"
+          }
+          ListElement {
+            type: "New Feature"
+            description: "Full snapping support providing snapping results and Z values of snapped feature"
+          }
+          ListElement {
+            type: "New Feature"
+            description: "Login for WMS / WFS layers"
```
Comment (m-kuhn) [inline src/qml/Changelog.qml]:
```suggestion
            description: "Login dialog for WMS and WFS layers"
```

What about postgres?

File: src/qml/Changelog.qml [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    id: mainGrid
+
+    anchors.fill: parent
+    anchors.margins: 20 * dp
+
+    columns: 1
+
+    Text {
+      id: title
+      text: qsTr( "What's new on QField %1").arg( version )
```
Comment (m-kuhn) [inline src/qml/Changelog.qml]:
```suggestion
      text: qsTr( "What's new in QField %1").arg( version )
```

**Comment by signedav** [comment]
Added:

- Font size like nirvn proposed
- word wrap in changelog list
- margins in changelog list
- title in green but without shade because I don't like it (tell me if I should add it still)
- expiring date

missing:
- [ ] jumps back on flick
- [ ] spaces below the title and the changeloglist cannot be displayed because the image takes the empty space...

File: src/qml/qgismobileapp.qml [hunk 3]
```diff
@@ -1045,6 +1045,46 @@ ApplicationWindow {
       __projectSource = platformUtilities.openProject()
     }
   }
+
+  Popup {
+    id: changelogPopup
+    parent: ApplicationWindow.overlay
+
+    property date expireDate: new Date(2019,8,9)
+    visible: settings.value( "/QField/CurrentVersion", "" ) !== versionCode+"test" && expireDate>new Date()
```
Comment (m-kuhn) [inline src/qml/qgismobileapp.qml]: Would a "remind me later" be possible too?

**Comment by signedav** [comment]
Screenshots from computer:
image

image

I'll check on device, when it's ready...

**Comment by signedav** [comment]
@nirvn you see the situation with the image. It takes all the space, and the spacer-rectangles between title and changelog-label and between changelog-label and crowdfunding-label are shrinked to zero.

I cannot make a minimum size for the spacer rectangle because then on landscape mode the image would be shrinked down... 

Maybe you have an idea here..?

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-95d7472289be08cf3047b310c7e20d26072f6382-armv7-X.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-95d7472289be08cf3047b310c7e20d26072f6382-x86-X.apk) (Android 5: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-95d7472289be08cf3047b310c7e20d26072f6382-armv7-5.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-95d7472289be08cf3047b310c7e20d26072f6382-x86-5.apk))

**Comment by signedav** [comment]
Screenshot_20190926-143415_QField Beta 624
Screenshot_20190926-143424_QField Beta 624

**Comment by signedav** [comment]
Like this on tablet

Screenshot_20190926-144155_QField Beta 624
Screenshot_20190926-144206_QField Beta 624

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-7749b9ea95b40d58d976d93d99170993d7cebaee-armv7-X.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-7749b9ea95b40d58d976d93d99170993d7cebaee-x86-X.apk) (Android 5: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-7749b9ea95b40d58d976d93d99170993d7cebaee-armv7-5.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-7749b9ea95b40d58d976d93d99170993d7cebaee-x86-5.apk))

**Comment by 3nids** [comment]
I think the texts in the button should be bigger and maybe bolder.
And maybe maybe white on green instead of black on green.

**Comment by signedav** [comment]
> @nirvn you see the situation with the image. It takes all the space, and the spacer-rectangles between title and changelog-label and between changelog-label and crowdfunding-label are shrinked to zero.
> 
> I cannot make a minimum size for the spacer rectangle because then on landscape mode the image would be shrinked down...
> 
> Maybe you have an idea here..?

Fixed that by having on portrait mode maximum as height a third of the width. It's a little dirty but looks better...

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-a6916dc8bb182298ff95283387935da84e2fbe42-armv7-X.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-a6916dc8bb182298ff95283387935da84e2fbe42-x86-X.apk) (Android 5: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-a6916dc8bb182298ff95283387935da84e2fbe42-armv7-5.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-a6916dc8bb182298ff95283387935da84e2fbe42-x86-5.apk))

**Comment by signedav** [comment]
image
Like that @3nids ?

**Comment by 3nids** [comment]
> Like that @3nids ?
yeah, much better to me, cool!

**Comment by m-kuhn** [comment]
Can we make the changelog fill up the white space in between?

**Comment by signedav** [comment]
> Can we make the changelog fill up the white space in between?

I'll check...

**Comment by m-kuhn** [comment]
How about changing the wording to:

"Want to support QField? Support the crowdfunding project before October 9."

**Comment by signedav** [comment]
image
image

good idea @m-kuhn 


for the changelog I'll add a minimum size:
image

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-c4ffb7b7f56c4fc8f8eb9a198a2b6879d51701e3-armv7-X.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-c4ffb7b7f56c4fc8f8eb9a198a2b6879d51701e3-x86-X.apk) (Android 5: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-c4ffb7b7f56c4fc8f8eb9a198a2b6879d51701e3-armv7-5.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-c4ffb7b7f56c4fc8f8eb9a198a2b6879d51701e3-x86-5.apk))

**Comment by signedav** [comment]
> "Want to support QField? Support the crowdfunding project before October 9."

Or, "Do you love QField? Support the crowdfunding project before October 9."
@m-kuhn

**Comment by nirvn** [comment]
Another suggestion: I'd get rid of the release nickname (Matterhorn), and even possibly the version number. Keep it lean and clean. "What's new in the latest QField"

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-5c77f557264b8a68037c2b7746ca97fb8a8db599-armv7-X.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-5c77f557264b8a68037c2b7746ca97fb8a8db599-x86-X.apk) (Android 5: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-5c77f557264b8a68037c2b7746ca97fb8a8db599-armv7-5.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-5c77f557264b8a68037c2b7746ca97fb8a8db599-x86-5.apk))

**Comment by m-kuhn** [comment]
I'd probably also drop the frame around the changelog. Or make it gray.

**Comment by signedav** [comment]
> I'd probably also drop the frame around the changelog. Or make it gray.

It is gray. 

Matthieu proposed it as well. But if we remove it it does look less like a scrollable list ...

**Comment by m-kuhn** [comment]
can the gray be done lighter then?

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-bfd25e3969f4e7e8ff7e0147e6738db5fd4eff8e-armv7-X.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-bfd25e3969f4e7e8ff7e0147e6738db5fd4eff8e-x86-X.apk) (Android 5: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-bfd25e3969f4e7e8ff7e0147e6738db5fd4eff8e-armv7-5.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-bfd25e3969f4e7e8ff7e0147e6738db5fd4eff8e-x86-5.apk))

**Comment by signedav** [comment]
- cared about your inputs
- made new button for remind me later and decline
- inserted link

image
image

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-2025483c37c8492e92484b0bc23108e9f7011a86-armv7-X.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-2025483c37c8492e92484b0bc23108e9f7011a86-x86-X.apk) (Android 5: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-2025483c37c8492e92484b0bc23108e9f7011a86-armv7-5.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-2025483c37c8492e92484b0bc23108e9f7011a86-x86-5.apk))

File: src/qml/Changelog.qml [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+        }
+        focus: true
+        clip: true
+        highlightRangeMode: ListView.StrictlyEnforceRange
+      }
+    }
+
+    Text {
+      color: '#90000000'
+      text: qsTr( "Do you love QField? Support the crowdfunding project before October 9." )
```
Comment (m-kuhn) [inline src/qml/Changelog.qml]:
```suggestion
      text: qsTr( "Do you love QField? Support the crowdfunding campaign. Before October 9." )
```

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-f32b7ffcc7a676896317e05b2a3435789b44db89-armv7-X.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-f32b7ffcc7a676896317e05b2a3435789b44db89-x86-X.apk) (Android 5: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-f32b7ffcc7a676896317e05b2a3435789b44db89-armv7-5.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-f32b7ffcc7a676896317e05b2a3435789b44db89-x86-5.apk))

**Comment by m-kuhn** [comment]
Peek 2019-09-26 18-30

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-679ab6254f850fb036057f185be7974da99181ca-armv7-X.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-679ab6254f850fb036057f185be7974da99181ca-x86-X.apk) (Android 5: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-679ab6254f850fb036057f185be7974da99181ca-armv7-5.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-679ab6254f850fb036057f185be7974da99181ca-x86-5.apk))

**Comment by mbernasocchi** [comment]
looks nice!

**Comment by mbernasocchi** [comment]
on my device it doesn't look as nice as here. 
Screenshot_20190926-184425_QField Beta 624
it needs more padding at the top. and I still think the white box should have rounded corners like the unpacking dialog or is there a reason why you keep on ignoring me on this 😬

**Comment by mbernasocchi** [comment]
also the url at the bottom can be opengis.ch/qfield-love

**Comment by m-kuhn** [comment]
@mbernasocchi does it work for you to click No / Maybe Later?
It always reappears for me, no matter what I click

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-30c9f2486f8795fd0708bd135e87bf9633d758be-armv7-X.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-30c9f2486f8795fd0708bd135e87bf9633d758be-x86-X.apk) (Android 5: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-30c9f2486f8795fd0708bd135e87bf9633d758be-armv7-5.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-30c9f2486f8795fd0708bd135e87bf9633d758be-x86-5.apk))

**Comment by VxTedxV** [comment]
> Another suggestion: I'd get rid of the release nickname (Matterhorn), and even possibly the version number. Keep it lean and clean. "What's new in the latest QField"

I think the version number should remain in order to know which version is the one installed with the improvements described.

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-8cb16c63c90a0b215df5303c9b2b16aba5751db3-armv7-X.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-8cb16c63c90a0b215df5303c9b2b16aba5751db3-x86-X.apk) (Android 5: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-8cb16c63c90a0b215df5303c9b2b16aba5751db3-armv7-5.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-624-8cb16c63c90a0b215df5303c9b2b16aba5751db3-x86-5.apk))

**Comment by Emic37** [comment]
Just tested now. It looks good for me. And all buttons works well. If you had the strings to transifex, I can translate that before update. 😉

**Comment by m-kuhn** [comment]
I just merged this to get the strings into transifex.

Please followup with additional feedback where appropriate, it's much appreciated still!

**Comment by signedav** [comment]
Thanks @m-kuhn

Sorry I forgot about your round corner input @mbernasocchi I could implement them on Monday if required. But I think it's preferable to release it asap otherwise the call for crowdfunding makes no sense anymore.

If we will keep this chanchelog dialog (what I would suggest) in future releases, we can IMO invest on further finetuning. Then I would like to have as well the combination of dynamic sized elements and the scrollable dialoge to avoid having white space on portrait and less need to scroll on landscape.

But for the moment I think it's okay as it is.

**Comment by mbernasocchi** [comment]
@m-kuhn what about adding 1 week to the crowdfunding

**Comment by m-kuhn** [comment]
@Emic37 and everyone, translation ready

**Comment by mbernasocchi** [comment]
@nirvn what do you think about inverting the colors in the title and do white text on a green background to "close off" the top of the dialog (see #634)

**Comment by Emic37** [comment]
@m-kuhn thanks. Do the buttons resize themselves automatically? 
Maybe this answer is stipid, but "no" in French is "non" and there could be not enough space if it isn't the case. It could be the same issue for other languages.

**Comment by m-kuhn** [comment]
Yes, they do resize automatically, they even support the German "Nein" :)
Actually, I wanted to make them 50% width, but still need to find out how exactly.

**Comment by Emic37** [comment]
Ok. I was thinking about German too... My question was stupid! 😁

**Comment by m-kuhn** [comment]
> Ok. I was thinking about German too... My question was stupid! 😁

Now, that's you who said that :wink:

**Comment by signedav** [comment]
> Actually, I wanted to make them 50% width, but still need to find out how exactly.

I like it when the No is smaller. It's like "You can say no, but we do not suggest you to say no" with an eye wink. But 50 % is maybe more harmonic.

**Comment by mbernasocchi** [comment]
i prefer smaller no. let's leave it that way

**Pull request merged**